File: gto-support-db/src/testCommon/java/org/ccci/gto/android/common/db/RootMapper.java
package org.ccci.gto.android.common.db;

import android.content.ContentValues;
import android.database.Cursor;
import androidx.annotation.NonNull;

import org.ccci.gto.android.common.db.model.Root;
import org.ccci.gto.android.common.db.util.CursorUtils;

import static org.ccci.gto.android.common.db.Contract.RootTable.COLUMN_ID;
import static org.ccci.gto.android.common.db.Contract.RootTable.COLUMN_TEST;

class RootMapper extends AbstractMapper<Root> {
    @Override
    protected void mapField(@NonNull ContentValues values, @NonNull String field, @NonNull Root obj) {
        switch (field) {
            case COLUMN_ID:
                values.put(field, obj.id);
                break;
            case COLUMN_TEST:
                values.put(field, obj.test);
                break;
            default:
                super.mapField(values, field, obj);
                break;
        }
    }

    @NonNull
    @Override
    protected Root newObject(@NonNull Cursor c) {
        return new Root();
    }

    @NonNull
    @Override
    public Root toObject(@NonNull Cursor c) {
        Root root = super.toObject(c);

        root.id = CursorUtils.getLong(c, COLUMN_ID);
        root.test = CursorUtils.getString(c, COLUMN_TEST);

        return root;
    }
}

File: gto-support-db/src/testCommon/java/org/ccci/gto/android/common/db/RootMapper.kt
package org.ccci.gto.android.common.db

import android.content.ContentValues
import android.database.Cursor
import org.ccci.gto.android.common.db.Contract.RootTable.COLUMN_ID
import org.ccci.gto.android.common.db.Contract.RootTable.COLUMN_TEST
import org.ccci.gto.android.common.db.model.Root
import org.ccci.gto.android.common.db.util.CursorUtils
import org.ccci.gto.android.common.util.database.getString

internal class RootMapper : AbstractMapper<Root>() {
    override fun mapField(values: ContentValues, field: String, obj: Root) = when (field) {
        COLUMN_ID -> values.put(field, obj.id)
        COLUMN_TEST -> values.put(field, obj.test)
        else -> super.mapField(values, field, obj)
    }

    override fun newObject(c: Cursor) = Root()
    override fun toObject(c: Cursor) = super.toObject(c).apply {
        id = CursorUtils.getLong(c, COLUMN_ID)
        test = c.getString(COLUMN_TEST)
    }
}
